applications/SolidMechanicsApplication/solid_mechanics_application.cpp

```diff
@@ -869,8 +869,39 @@ void KratosSolidMechanicsApplication::Register() {
     KRATOS_REGISTER_CONDITION(
         "SurfaceElasticCondition3D9N", mSurfaceElasticCondition3D9N)
 
-    //Register Constitutive Laws
+    //Register Constitutive Laws with KRATOS_REGISTER to the Kernel
+    //Linear Elastic laws
```

Comment:
Why do you register here, are these laws registered on the solid?

Reply:
They are registered as serializer currently, but the read_material_process cant detect it unless it is registered in kratos kernel. I have confirmed this with @josep-m-carbonell yesterday.

Reply:
Nope, the problem is that in 

https://github.com/KratosMultiphysics/Kratos/blob/5cfb348eedd8d6fac56e8de65ac4bf69cbd8242b/applications/SolidMechanicsApplication/solid_mechanics_application.cpp#L875

Is not using the proper macro:

https://github.com/KratosMultiphysics/Kratos/blob/5cfb348eedd8d6fac56e8de65ac4bf69cbd8242b/applications/StructuralMechanicsApplication/structural_mechanics_application.cpp#L454

@josep-m-carbonell can we change that?, I feel unnecessary to create a dependency from one application to other

Reply:
I think that replacing  Serializer::Register for  KRATOS_REGISTER_CONSTITUTIVE_LAW  is ok. Just ensure to not duplicate both registers.

Reply:
Yes, the register is including the serialization.